packages/localize-tools/src/formatters/xlb.ts

```diff
@@ -90,7 +91,9 @@ class XlbFormatter implements Formatter {
           ) {
             throw new KnownError(`Expected <ph> to have exactly one text node`);
           }
-          contents.push({untranslatable: phText.nodeValue || ''});
+          const index =
+            Number((child as Element).getAttribute('name')) || phIdx++;
+          contents.push({untranslatable: phText.nodeValue || '', index});
```

Comment:
@aomarks this is an attempt for keeping existing xlb compatibility (dumbly assigning indexes as they appear, since matching by content seemed like a lot more work) but I wonder if it's actually more dangerous than throwing since it can silently mess up the order of the placeholders in the translation.

Reply:
Yeah, good thought, but I agree that it could actually be harmful because it would silently generate incorrect templates in the case that a placeholder was repositioned. I think it's fine to throw here.